### PR TITLE
Allow reviewing or reverting to a page's initial revision

### LINKED ISSUE #8337: Can't review or revert to a page's initial revision
### Issue Summary

Using the Page History view, there's no way to restore the very first revision of a page, whether it was first saved as draft, published, or submitted for moderation.

This is because the Page History view is populated from PageLogEntry objects, but Wagtail doesn't log the first revision of a page. Note `log_action=False` in these lines when a new page is saved, published, or submitted for moderation:

https://github.com/wagtail/wagtail/blob/629ced01cabaf0e24e5cbc70bf552b9fc046b5b0/wagtail/admin/views/pages/create.py#L172
https://github.com/wagtail/wagtail/blob/629ced01cabaf0e24e5cbc70bf552b9fc046b5b0/wagtail/admin/views/pages/create.py#L198
https://github.com/wagtail/wagtail/blob/629ced01cabaf0e24e5cbc70bf552b9fc046b5b0/wagtail/admin/views/pages/create.py#L254

A simple fix might be to change these all to `log_action=True`. It does appear that the original implementation [deliberately avoided](https://github.com/wagtail/wagtail/commit/2c2aa68503e798529abe6f86240116263803791f) creating an initial `wagtail.edit` event for new pages. Unfortunately that does seem to make it impossible to review or revert to a page's initial state.

Thanks to Slack user Gabriel for bringing up this issue in chat [here](https://wagtailcms.slack.com/archives/C81FGJR2S/p1649733822692019), and @cnk for helping with the explanation. Per @zerolab a workaround on versions up to 2.15 is to change the page history URL `admin/pages/ID/history/` to `admin/pages/ID/revisions/` to see the old revisions view, but this seems to have been removed in 2.16 in https://github.com/wagtail/wagtail/commit/6931663c7f8b419de77085b918329e23483edd45.

Another limited workaround is that users can "Compare with previous revision" on the 2nd revision of a page - this will at least display the list of field-by-field changes so editors can review the differences. But this is far less useful than being able to review or restore the entire revision.

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. `cd myproject`
3. `./manage.py migrate`
4. `./manage.py createsuperuser` and create an admin user.
5. `./manage.py runserver`
6. Vist http://localhost:8000/admin/pages/add/home/homepage/3/ to create a new page (you'll be asked to log in first)
7. Fill in the title field with something like "Original". Click "Save Draft".
8. Change the title field to something like "Changed". Click "Save Draft" again.
9. Now click on "History". You'll see there's no way to restore the original revision:

image

Alternatively, if you first publish the page instead of saving:

<ol start="7">
<li>Fill in the title field with something like "Original". Click "Publish" in the actions menu in the footer.</li>
<li>Click "Edit" to re-edit the page that was just created.</li>
<li>Change the title field to something like "Changed". Click "Save Draft".</li>
<li>Now click on "History". You'll see there's no way to restore the originally published version:</li>
</ol>

image

Alternatively, if you first submit the page for moderation:

<ol start="7">
<li>Fill in the title field with something like "Original". Click "Submit for moderation" in the actions menu in the footer.</li>
<li>Click "Edit" to re-edit the page that was just created.</li>
<li>Change the title field to something like "Changed". Click "Save Draft".</li>
<li>Now click on "History". You'll see there's no way to restore the originally published version:</li>
</ol>

image

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

* Python version: 3.8.12
* Django version: 4.0.4
* Wagtail version: 2.16.2

### PR DESCRIPTION
**Pull request by chosak**
Currently it's not possible to review or revert to a page's initial revision because we don't log a PageLogEntry "wagtail.edit" event in certain cases:

- When a page is first created, and saved as draft
- When a page is first created, and published right away
- When a page is first created, and submitted for moderation

This commit alters the current behavior so that a "wagtail.edit" event is created in those cases. This allows these initial revisions to be reviewed or reverted to in the page history view.

This commit also updates the existing create_log_entries_from_revisions management command to try to populate these initial revisions there as well. The relevant unit tests have been updated.

Fixes #8337 - please see that issue for specific testing steps.

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [x] For Python changes: Have you added tests to cover the new/fixed behaviour?

### CODE REVIEW

**Comment by th3hamm0r** [comment]
Thanks @chosak, this small change really fixes an annoying issue for us. I've just tested it with 4.1.1 and it looks good to me.
Beside the merge conflict, which is probably easy to fix, is there something holding us back to merge this? @gasman

**Comment by lb-** [comment]
@th3hamm0r you are a core team member now - feel free to do a formal review (using the GitHub review step) and merge this in.

I have not seen any chatter about this being blocked for any reason, I think it's just waiting on a core team review and then merging.

The code seems solid to me but I have not tested, but if you approve I am happy to do the merging/rebasing bit.

**Comment by lb-** [comment]
Just re-added the needs review label - it looks like I accidentally removed it the other week - I have not done a review on this.

**Comment by th3hamm0r** [comment]
Thanks @lb- ! I will do that soon. That was a missing bit to me about the workflow, since I'm of course not in the position to do any merging yet 😅

**Comment by th3hamm0r** [comment]
@chosak could you rebase this on main?
There will be one conflict here https://github.com/wagtail/wagtail/pull/8949/files#diff-c77699a2ec17cab0eaa4a20e3e39d1623e846091c8dc0abfc3aad7115a0627f9R228, where "wagtail.edit" needs to be added in alphabetical order (see #9611), so it should be:
`["wagtail.create", "wagtail.edit", "wagtail.publish"]`

**Comment by chosak** [comment]
@th3hamm0r rebased! Thanks for offering to review.

**Review by th3hamm0r — APPROVED** [review]
LGTM

I've tested the 3 ways of creating a page and reverting after changes with a local installation.
The management task has been created for the migration to wagtail >=2.10, so since the code looks good to me and the tests run, I would give that a go too ;)

**Pull request merged**
